lib/inventory/servicecounter.go

```diff
@@ -0,0 +1,63 @@
+/*
+Copyright 2023 Gravitational, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package inventory
+
+import (
+	"sync"
+	"sync/atomic"
+
+	"github.com/gravitational/teleport/api/types"
+)
+
+// serviceCounter will count services seen in the inventory.
+type serviceCounter struct {
+	countMap *sync.Map
+}
+
+// counts returns the count of each service seen in the counter.
+func (s *serviceCounter) counts() map[types.SystemRole]uint64 {
```

Comment:
Is it worth it to add a dedicated function to fetch the count for a single role?

Reply:
I was waffling about that. It makes sense, but it's also not a huge deal to just query the map. I'm ambivalent, I think. If you feel strongly I can add one.

Reply:
I just realized that `serviceCounter` has a `get` method that does that already (although it could be made to not actually add to the map if there's no need for it).

I wouldn't bother with wiring it up for the gRPC service (making a map is nothing compared to the overhead of protobuf encoding and network transmission, and saving like 5 strings and integers worth of data is kinda useless) but it might be useful as an exposed `ConnectedServiceCount(types.SystemRole)` method in `Controller`. The first usecase for this, after all, is going to be about checking if there's any `Okta` agents connected, and we won't care about any other count.